worlds/timespinner/__init__.py

```diff
@@ -102,19 +108,78 @@ def generate_basic(self):
     def fill_slot_data(self) -> Dict[str, object]:
         slot_data: Dict[str, object] = {}
 
+        ap_specific_settings: Set[str] = { "RisingTidesOverrides", "TrapChance" }
+
         for option_name in timespinner_options:
-            slot_data[option_name] = get_option_value(self.multiworld, self.player, option_name)
+            if (option_name not in ap_specific_settings):
+                slot_data[option_name] = get_option_value(self.multiworld, self.player, option_name)
 
         slot_data["StinkyMaw"] = True
         slot_data["ProgressiveVerticalMovement"] = False
         slot_data["ProgressiveKeycards"] = False
-        slot_data["PyramidKeysGate"] = self.pyramid_keys_unlock
         slot_data["PersonalItems"] = get_personal_items(self.player, self.location_cache)
+        slot_data["PyramidKeysGate"] = self.precalculated_weights.pyramid_keys_unlock
+        slot_data["PresentGate"] = self.precalculated_weights.present_key_unlock
+        slot_data["PastGate"] = self.precalculated_weights.past_key_unlock
+        slot_data["TimeGate"] = self.precalculated_weights.time_key_unlock
+        slot_data["Basement"] = int(self.precalculated_weights.flood_basement) + \
+                                int(self.precalculated_weights.flood_basement_high)
+        slot_data["Xarion"] = self.precalculated_weights.flood_xarion
+        slot_data["Maw"] = self.precalculated_weights.flood_maw
+        slot_data["PyramidShaft"] = self.precalculated_weights.flood_pyramid_shaft
+        slot_data["BackPyramid"] = self.precalculated_weights.flood_pyramid_back
+        slot_data["CastleMoat"] = self.precalculated_weights.flood_moat
+        slot_data["CastleCourtyard"] = self.precalculated_weights.flood_courtyard
+        slot_data["LakeDesolation"] = self.precalculated_weights.flood_lake_desolation
+        slot_data["DryLakeSerene"] = self.precalculated_weights.dry_lake_serene
 
         return slot_data
 
     def write_spoiler_header(self, spoiler_handle: TextIO):
-        spoiler_handle.write('Twin Pyramid Keys unlock:        %s\n' % (self.pyramid_keys_unlock))
+        if is_option_enabled(self.multiworld, self.player, "UnchainedKeys"):
+            spoiler_handle.write('Modern Warp Beacon unlock:       %s\n' % 
+                (self.precalculated_weights.present_key_unlock))
```

Comment:
Also the way you pass in the strings looks wrong:
Having `(something)` looks like it should be `(something,)` (i.e. a tuple). If you explicitly just want `something` (i.e. a str), then remove the `()` - sounds like a small thing, but having to figure out what a pair of `()` does makes reviewing a bit harder.